qutebrowser/mainwindow/tabwidget.py

```diff
@@ -517,7 +517,9 @@ def tabSizeHint(self, index):
 
             no_pinned_count = self.count() - self.pinned_count
             pinned_width = tab_width_pinned_conf * self.pinned_count
-            no_pinned_width = self.width() - pinned_width
+            # Prevent any tabs from being smaller than the min size
+            no_pinned_width = max(self.width() - pinned_width,
+                                  minimum_size.width() * no_pinned_count)
```

Comment:
Hmm, I didn't notice that (I really should have). The difference between handling this case in the block above and below is that are similar, but if I try to move the check into the block above, then it would set the size of pinned tabs to the min width. I'm going to try to merge the two if blocks to keep the existing behavior intact, and be more correct.